### DISCUSSION TITLE
Is it possible to Shuffle() without deleting items?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#894] [Q&A] [answered]
Hi

While using shuffle(), I saw that it destroys values... 

[codepen here](https://codepen.io/galinhalx/pen/YzaVozL)

Is there a way to mix values, without destroying them?

(Or is this a bug?)

**Comment by DmitrySharabin** [comment, marked as answer]
Hey there,

It looks like the `mv-action="set(myList, shuffle(myList))"` expression leads to some sort of race condition. Try using a computed property instead, like so:

```
<meta property="foo" content="[shuffle(myList)]" />

<button mv-action="set(myList, foo)">Shuffle (press multiple times)</button>
```

**Reply by GalinhaLX** [reply]
Hi

There's a strange effect with @DmitrySharabin 's solution:
-> When number of values are low (e.g. 3) , shuffle() stops to work after a few clicks

sounds like a bug, no? 

[codepen here](https://codepen.io/galinhalx/pen/NWYaWde)
